torchreid/losses/am_softmax.py

```diff
@@ -175,7 +175,7 @@ def forward(self, cos_theta, target, aug_index=None, lam=None, iteration=None):
             iteration (int): current iteration
         """
         if (self.aug_type and aug_index is not None and lam is not None):
-            targets1 = torch.zeros(cos_theta.size()).scatter_(1, target.unsqueeze(1).data.cpu(), 1)
+            targets1 = torch.zeros(cos_theta.size(), device=target.device).scatter_(1, target.detach().unsqueeze(1), 1)
             if self.use_gpu:
                 targets1 = targets1.cuda()
```

Comment:
If `target` is on GPU then `targets1` too. So there is no need to use `.cuda()`.